File: 61_Rotate_List.js
var rotateRight = function(head, k) {
    if(!head || !head.next || k===0)return head
    let l =1
    let tail = head;
    while(tail.next){
        tail = tail.next;
         l++
    }
 k = k%l
    if(k===0 )return head
    tail.next = head 
    let b = l-k
    let n = head
    for (let i =1;i<b;i++)n = n.next
    let nh = n.next 
    n.next = null
    return nh
};

File: 61_Rotate_List.rs
impl Solution {
    pub fn rotate_right(
        mut head: Option<Box<ListNode>>, 
        k: i32
    ) -> Option<Box<ListNode>> {

        if head.is_none() {
            return head;
        }

        let mut length = 0;
        let mut current = head.as_ref();

        while let Some(node) = current {
            length += 1;
            current = node.next.as_ref();
        }

        let k = (k as usize) % length;
        if k == 0 {
            return head;
        }

        let breaking = length - k;

        let mut current = head.as_mut();

        for _ in 0..breaking - 1 {
            current = current.unwrap().next.as_mut();
        }

        let mut start_node = current.as_mut().unwrap().next.take();

        let mut tail = start_node.as_mut();
        while let Some(node) = tail {
            if node.next.is_none() {
                node.next = head;
                break;
            }
            tail = node.next.as_mut();
        }

        start_node
    }
}